### ISSUE TITLE
Unexpected behavior with links to System32 and SysWOW64 in Syscheck

### ISSUE THREAD

**Issue by crolopez** [wazuh/wazuh#3739]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.3 | Syscheck and Logcollector | Agent | Packages and sources | Windows Server 2008 R2 |

To monitor directories in Syscheck or files in Logcollector from System32 or SysWOW64 is needed to change the path as follows:

If we want to monitor `C:\Windows\System32\folder` with Syscheck in a 64-bit Windows, you have to replace `System32` by Sysnative:

``` XML
<syscheck>
<directories check_all="yes" realtime="yes">C:\Windows\Sysnative\folder</directories> 
</syscheck>
```

If the folder is `C:\Windows\SysWOW64\folder`, we have to replace `SysWOW64` by `System32`:

``` XML
<syscheck>
<directories check_all="yes" realtime="yes">C:\Windows\System32\folder</directories> 
</syscheck>
```
If the system is 32 bits, the configured path will not change. The same for Logcollector files.

The problem appears with some directories which do not follow this rule. For example, if we want to monitor `C:\Windows\System32\drivers\etc` in a 32 or 64 bits system, we have to set that path literally in the Syscheck block.

``` XML
<syscheck>
<directories check_all="yes" realtime="yes">C:\Windows\System32\drivers\etc</directories> 
</syscheck>
```

It seems that `C:Windows\System32` does not point to `C:Windows\SysWOW64` in these cases. 

Also, due to the automatic path conversion of whodata for Windows, **Wazuh is not able to process events in whodata mode for this type of folders**.

In short, if we want to monitor most of `C:\Windows\System32` subdirectories in 64-bits, we will have to replace `system32` with `sysnative` in the Syscheck configuration. In the case of `C:\Windows\SysWOW64` subdirectories, we will have to use `system32`. But if we find a directory that does not follow this rule like `C:\Windows\System32\drivers\etc`, we will have to indicate it literally in the configuration and not monitor it with whodata.

We have to look for information about this type of directories that do not follow the typical System32/SysWOW64 link. Wazuh should automatically detect the real path (linked or not) of these files/folders.

**Comment by FernandoCP** [comment]
Hi team! 

I found the list of exceptions of directories that don't follow the rule and it's this:

- %windir%\system32\catroot

- %windir%\system32\catroot2

- %windir%\system32\driverstore

- %windir%\system32\drivers\etc

- %windir%\system32\logfiles

- %windir%\system32\spool

**Windows Server 2008, Windows Vista, Windows Server 2003 and Windows XP:  **%windir%\system32\driverstore is redirected.

Attached here is the Microsoft documentation page from which I got the information:
https://docs.microsoft.com/en-us/windows/win32/winprog64/file-system-redirector

Best regards, Fernando

**Comment by vikman90** [comment]
Development complete.
Pinging @bah07 and his team to update the integrations tests and merge the PR.

**Comment by chemamartinez** [comment]
Closing this issue since it is outdated. This problem has been addressed at https://github.com/wazuh/wazuh/issues/14796.
